dash_auth/oidc_auth.py

```diff
@@ -274,6 +284,10 @@ def callback(self, idp: str):  # pylint: disable=C0116
             if self.log_signins:
                 logging.info("User %s is logging in.", user.get("email"))
 
+        additional_params = self._additional_session_params.get(idp)
+        if additional_params:
+            session["user"].update(additional_params)
```

Comment:
Instead of just a static dict of params per idp it would be good to pass a function to OIDCAuth that takes as argument `user` from line 278 and `idp` to update `user`.

Maybe the best option would be to separate the block from l.279 to the return into its own method that can be overridden/monkeypatched by users if required.

Something like
```py
        user = token.get("userinfo")
        return self.after_logged_in(user, idp)

    def after_logged_in(self, user: dict | None, idp: str):
        if user:
            session["user"] = user
            # ... etc like the above

        return redirect(self.app.config.get("url_base_pathname") or "/")
```

This would allow someone to extend this downstream like:
```py
class MyOIDCAuth(OIDCAuth):
    def after_logged_in(self, user: dict | None, idp: str):
        if user:
            user["other_param"] = my_func(user["email"])

        return super().after_logged_in(user, idp)
```